freshmaker/handlers/botas/botas_shipped_advisory.py

```diff
@@ -66,11 +71,75 @@ def handle(self, event):
             "Orignial nvrs of build in the advisory #{0} are: {1}".format(
                 db_event.search_key, " ".join(original_nvrs)))
 
-        msg = "Skipping due to being blocked on further implementation for now."
+        index_images = self._pyxis.get_operator_indices()
+        # get latest bundle images per channel per index image filtered by higher version
+        all_bundles = self._pyxis.get_latest_bundles(index_images)
+        bundles = self._pyxis.filter_bundles_by_nvrs(original_nvrs, all_bundles)
+        bundle_digests = []
+        for bundle in bundles:
+            if not bundle.get('bundle_path_digest'):
+                log.warning(f"Bundle {bundle['bundle_path']} doesn't have"
+                            " 'bundle_path_digests' set")
+                continue
+            bundle_digests.append(bundle['bundle_path_digest'])
+        bundle_images = self._pyxis.get_images_by_digests(bundle_digests)
+
+        # now we have nvrs of bundles to rebuild
+        bundles_nvrs = self._filter_bundle_images_by_builds(bundle_images)  # noqa: F841
 
         # Skip that event because we can't proceed with processing it.
         # TODO
-        # Next step would be queries Pyxis and getting the digests of these nvrs
+        # Now when we have bundle images' nvrs we can procceed with rebuilding it
+        msg = "Skipping due to being blocked on further implementation for now."
         db_event.transition(EventState.SKIPPED, msg)
         db.session.commit()
         return []
+
+    def _filter_bundle_images_by_builds(self, bundle_images):
+        """
+        Filter out bundle images that don't have OSBS-pinning
+
+        :param dict bundle_images: Mapping of bundle digest to list of bundle's
+        ContainerImages
+        :return: list of nvr's of bundle images that have OSBS-pinning
+        """
+        ret_bundle_images_nvrs = set()
+        with koji_service(conf.koji_profile, log, dry_run=conf.dry_run,
+                          login=False) as session:
+            for digest, bundles in bundle_images:
+                for nvr in [bundle['brew']['build'] for bundle in bundles]:
+                    build = session.get_build(nvr)
+                    related_images = (
+                        build.get("build", {})
+                             .get("extra", {})
+                             .get("image", {})
+                             .get("operator_manifests", {})
+                             .get("related_images", {})
+                    )
+                    # skip bundle if it's not created by OSBS
+                    if not related_images.get("created_by_osbs", False):
+                        continue
+
+                    for pullspec in related_images.get('pullspecs', []):
+                        related_image_digest = self.parse_digest(pullspec['new'])
+                        if digest != related_image_digest:
```

Comment:
This doesn't look right since `digest` is the manifest list of the bundle image. It should be comparing to see if the digest of the related image matches the operator/operand image that was rebuilt by Freshmaker which triggered the event for rebuilding the bundle images.

Reply:
You mean comparing `repositories.manifest_list_digest` of the rebuilt operator/operand images and digests of the related images?

Reply:
Yes, comparing `manifest_list_digest` of the operator/operand image that needed to be rebuilt and not the rebuilt one itself.